nex/matchmake-extension/update_notification_data.go

```diff
@@ -1,32 +1,41 @@
 package nex_matchmake_extension
 
 import (
-	"fmt"
+	"log"
 
 	nex "github.com/PretendoNetwork/nex-go"
 	"github.com/PretendoNetwork/wiiu-chat-secure/database"
 	"github.com/PretendoNetwork/wiiu-chat-secure/globals"
+	"github.com/PretendoNetwork/wiiu-chat-secure/grpc"
 	nex_notifications "github.com/PretendoNetwork/wiiu-chat-secure/nex/notifications"
 
 	matchmake_extension "github.com/PretendoNetwork/nex-protocols-go/matchmake-extension"
 )
 
 func UpdateNotificationData(err error, client *nex.Client, callID uint32, uiType uint32, uiParam1 uint32, uiParam2 uint32, strParam string) {
-	// TODO: Implement this
-	fmt.Printf("uiType: %d, uiParam1: %d, uiParam2: %d, strParam: %s\r\n", uiType, uiParam1, uiParam2, strParam)
+	log.Printf("uiType: %d, uiParam1: %d, uiParam2: %d, strParam: %s\r\n", uiType, uiParam1, uiParam2, strParam)
+
+	if uiType == 101 {

```

Comment:
Only gonna comment this once despite it being elsewhere

Do we know if these types are specific to WiiU Chat? Or are they common to the matchmaking protocol as a whole?

If they're common to the protocol then I'm thinking we should make a struct for them there just like the notifications types

If they're specific to WiiU Chat, then a struct for their types should be made here instead, for clarity's sake so it's clear at a glance what each types value means

Reply:
> If they're common to the protocol then I'm thinking we should make a struct for them there just like the notifications types

In case this is common to the protocol (which I think it is), we don't have to make a special struct. Technically our documentation about notification types is *wrong*, because we store the whole number without splitting anything `109000`. However, [other Rendez-Vous servers](https://github.com/zeroKilo/GROBackendWV/wiki/Notifications) document notification types as `type * 1000 + subtype` (for example `109 * 1000 + 0`).

In this function we get the type without the subtype `101`, which is equivalent to the notification type `101000` (this notification looks like an invitation to join a gathering?)

Reply:
I forgot about this! Our documentation and implementations need to be updated then

Reply:
`nex-protocols-go` v1.0.36 has been released which changes how notification types are handled. Types are now split into categories and subtypes, the way they are implemented in other Rendezvous servers

This release also includes a helper function `BuildNotificationType`, which takes in a category and a subtype and returns `(category * 1000) + subtype`

Please update the notifications handling to use this new release